### PR TITLE
Remove jQuery from the create/rename branch modals (except Fomantic)

### PR DESCRIPTION
**Pull request by yardenshoham**
- Switched to plain JavaScript
- Tested the create/rename branch modals' functionality and they work as before

# Demo using JavaScript without jQuery
demo

### CODE REVIEW

File: web_src/js/features/repo-branch.js [hunk 1]
```diff
@@ -8,35 +8,41 @@ export function initRepoBranchButton() {
...
-    }
+  for (const element of document.querySelectorAll('.show-create-branch-modal')) {
+    element.addEventListener('click', () => {
+      let modalFormName = element.getAttribute('data-modal-form');
+      if (!modalFormName) {
+        modalFormName = '#create-branch-form';
+      }
+      const modalForm = document.querySelector(modalFormName);
+      if (!modalForm) return;
+      modalForm.action = modalForm.getAttribute('data-base-action') + element.getAttribute('data-branch-from-urlcomponent');
```
Comment (silverwind) [inline web_src/js/features/repo-branch.js]:
```suggestion
      modalForm.action = `${modalForm.getAttribute('data-base-action')}${element.getAttribute('data-branch-from-urlcomponent')}`;
```

File: web_src/js/features/repo-branch.js [hunk 2]
```diff
@@ -8,35 +8,41 @@ export function initRepoBranchButton() {
...
-    let modalFormName = $(this).attr('data-modal-form');
-    if (!modalFormName) {
-      modalFormName = '#create-branch-form';
-    }
-    $(modalFormName)[0].action = $(modalFormName).attr('data-base-action') + $(this).attr('data-branch-from-urlcomponent');
-    let fromSpanName = $(this).attr('data-modal-from-span');
-    if (!fromSpanName) {
-      fromSpanName = '#modal-create-branch-from-span';
-    }
+  for (const element of document.querySelectorAll('.show-create-branch-modal')) {
```
Comment (silverwind) [inline web_src/js/features/repo-branch.js]: Prefer `el` to `element`.

File: web_src/js/features/repo-branch.js [hunk 3]
```diff
@@ -8,35 +8,41 @@ export function initRepoBranchButton() {
...
 }
 
 function initRepoRenameBranchButton() {
-  $('.show-rename-branch-modal').on('click', function () {
-    const target = $(this).attr('data-modal');
-    const $modal = $(target);
-
-    const oldBranchName = $(this).attr('data-old-branch-name');
-    $modal.find('input[name=from]').val(oldBranchName);
+  for (const element of document.querySelectorAll('.show-rename-branch-modal')) {
```
Comment (silverwind) [inline web_src/js/features/repo-branch.js]: Same here with `el`.

**Review by silverwind — APPROVED** [review]

File: web_src/js/features/repo-branch.js [hunk 4]
```diff
@@ -8,35 +8,41 @@ export function initRepoBranchButton() {
...
-    let fromSpanName = $(this).attr('data-modal-from-span');
-    if (!fromSpanName) {
-      fromSpanName = '#modal-create-branch-from-span';
-    }
+  for (const el of document.querySelectorAll('.show-create-branch-modal')) {
+    el.addEventListener('click', () => {
+      let modalFormName = el.getAttribute('data-modal-form');
+      if (!modalFormName) {
+        modalFormName = '#create-branch-form';
+      }
```
Comment (delvh) [inline web_src/js/features/repo-branch.js]:
```suggestion
      let modalFormName = el.getAttribute('data-modal-form') || '#create-branch-form';
```

File: web_src/js/features/repo-branch.js [hunk 5]
```diff
@@ -8,35 +8,41 @@ export function initRepoBranchButton() {
...
+      if (!modalForm) return;
+      modalForm.action = `${modalForm.getAttribute('data-base-action')}${el.getAttribute('data-branch-from-urlcomponent')}`;
 
-    $(fromSpanName).text($(this).attr('data-branch-from'));
-    $($(this).attr('data-modal')).modal('show');
-  });
+      let fromSpanName = el.getAttribute('data-modal-from-span');
+      if (!fromSpanName) {
+        fromSpanName = '#modal-create-branch-from-span';
+      }
```
Comment (delvh) [inline web_src/js/features/repo-branch.js]:
```suggestion
      let fromSpanName = el.getAttribute('data-modal-from-span') || '#modal-create-branch-from-span';
```

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**